src/main/java/net/glasslauncher/annoyancefix/mixin/MinecraftMixin.java

```diff
@@ -0,0 +1,129 @@
+package net.glasslauncher.annoyancefix.mixin;
+
+import net.minecraft.block.BlockBase;
+import net.minecraft.client.Minecraft;
+import net.minecraft.entity.Boat;
+import net.minecraft.entity.EntityBase;
+import net.minecraft.entity.Minecart;
+import net.minecraft.entity.Painting;
+import net.minecraft.entity.player.AbstractClientPlayer;
+import net.minecraft.item.ItemBase;
+import net.minecraft.level.Level;
+import net.minecraft.util.hit.HitResult;
+import net.minecraft.util.hit.HitType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Map;
+
+import static java.util.Map.entry;
+
+@Mixin(value = Minecraft.class)
+public class MinecraftMixin {
+
+    @Unique
+    private final Map<Integer, Integer> pickBlockLookupMap = Map.ofEntries(
+            entry(BlockBase.REDSTONE_DUST.id, ItemBase.redstoneDust.id),
+            entry(BlockBase.REDSTONE_REPEATER.id, ItemBase.redstoneRepeater.id),
+            entry(BlockBase.REDSTONE_REPEATER_LIT.id, ItemBase.redstoneRepeater.id),
+            entry(BlockBase.WOOD_DOOR.id, ItemBase.woodDoor.id),
+            entry(BlockBase.IRON_DOOR.id, ItemBase.ironDoor.id),
+            entry(BlockBase.STANDING_SIGN.id, ItemBase.sign.id),
+            entry(BlockBase.WALL_SIGN.id, ItemBase.sign.id),
+            entry(BlockBase.CROPS.id, ItemBase.seeds.id),
+            entry(BlockBase.BED.id, ItemBase.bed.id),
+            entry(BlockBase.CAKE.id, ItemBase.cake.id)
+    );
+
+
+    @Shadow public HitResult hitResult;
+    @Shadow public Level level;
+    @Shadow public AbstractClientPlayer player;
+
+    // method_2103 is the pick block method
+    @Inject(at = @At("HEAD"), method = "method_2103", cancellable = true)
+    public void method_2103(CallbackInfo ci) {
+        int itemID = 0;
+
+        if (this.hitResult != null) {
+            // field_790 means "Entity"
```

Comment:
Great use of comments here! I saw in paulevs discord that he has some custom mapping that merge plasma mappings with the StationAPI mappings to define more field/method names. But, its probably too much effort for us to do. Love the comment.